presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkQueryExecutionFactory.java

```diff
@@ -819,4 +821,25 @@ private boolean isFatalException(Throwable t)
     {
         return t instanceof PrestoSparkFatalException;
     }
+
+    static Session.SessionBuilder transferSessionPropertiesToSession(Session.SessionBuilder session, Map<String, String> sessionProperties)
+    {
+        sessionProperties.forEach((key, value) -> {
+            // Presto session properties may also contain catalog properties in format catalog.property_name=value
+            String[] parts = key.split("\\.");
+            if (parts.length == 1) {
+                // session property
```

Comment:
session property -> system property
 (both are session properties, but some are system session properties and some are catalog session properties)